### PR TITLE
Setup api

### PR DESCRIPTION
**Pull request by yanqiuzhang**
#### Description
_As a development team
In order for the users to use the mobile application,
We need to set up back-end api_

#### Changes proposed in this pull request:
  * Create controller
 *  Add routes 
 * Requests api articles

### CODE REVIEW

File: spec/requests/list_all_articles_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+describe "Get /api/articles" do
+    let(:category){create(:category)}
+    let!(:articles){ 5.times {FactoryBot.create(:article)}}
+    let(:headers){ {HTTP_ACCEPT: "application/json"} }
+    
+    it "returns a collection of 5 articles" do
+        get '/api/articles', headers: headers
+    
+        expect(JSON.parse(response.body).count).to eq 5
```
Comment (magnus-thor) [inline spec/requests/list_all_articles_spec.rb]: write a helper file for parsing the response and then you can write this expect block like `expect(response_json.count).to eq 5`

for the helper file 
create a file `spec/support/response_json.rb` and add this in
```
module ResponseJSON
  def response_json
    JSON.parse(response.body)
  end
end

RSpec.configure do |config|
  config.include ResponseJSON
end
```
and add `Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }` to rails_helper.rb

**Review by magnus-thor — CHANGES_REQUESTED** [review]

File: spec/requests/list_all_articles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+describe "Get /api/articles" do
+    let(:category){create(:category)}
+    let!(:articles){ 5.times {FactoryBot.create(:article)}}
```
Comment (magnus-thor) [inline spec/requests/list_all_articles_spec.rb]: Create an article that is not published to make sure that you are only getting approved articles.

File: spec/requests/list_all_articles_spec.rb (hunk 1 above)
Comment (yanqiuzhang) [reply spec/requests/list_all_articles_spec.rb]: done

File: spec/requests/list_all_articles_spec.rb (hunk 2 above)
Comment (yanqiuzhang) [reply spec/requests/list_all_articles_spec.rb]: done

**Review by NiciP — APPROVED** [review]

**Review by magnus-thor — APPROVED** [review]

**Pull request merged**